clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/DeleteAmazonSnapshotAtomicOperation.java

```diff
@@ -47,9 +57,20 @@ private static Task getTask() {
   @Override
   public Void operate(List priorOutputs) {
     getTask().updateStatus(BASE_PHASE, String.format("Initializing Delete Snapshot operation for %s", description));
-    amazonClientProvider
-      .getAmazonEC2(description.getCredentials(), description.getRegion())
-      .deleteSnapshot(new DeleteSnapshotRequest().withSnapshotId(description.getSnapshotId()));
+    try {
+      amazonClientProvider
+        .getAmazonEC2(description.getCredentials(), description.getRegion())
+        .deleteSnapshot(new DeleteSnapshotRequest().withSnapshotId(description.getSnapshotId()));
+    } catch (AmazonEC2Exception ec2Exception) {
+      registry.counter(deleteSnapshotTaskId.withTag("success", false)).increment();
+      log.error("Failed to delete snapshotId with exception: {}", ec2Exception.getMessage());
+      throw ec2Exception;
+    } catch (Exception e) {
+      registry.counter(deleteSnapshotTaskId.withTag("success", false)).increment();
+      log.error("Failed to delete snapshotId with exception: {}", e.getMessage());
+      throw e;
```

Comment:
i don't think there is anything diff in these two handlers, did you intend to have some different behavior here? if not, can just leave the `Exception` one

Reply:
Ditto

Reply:
agree , updated it to just handle ```Exception```

Reply:
I should've noted this earlier... Might be useful to log the `snapshotId` to the `log.error`

Reply:
the exception message has the details of the ```snapshotId``` associated ```ami``` etc